blockly/src/utils/Direction.java

```diff
@@ -0,0 +1,86 @@
+package utils;
+
+import core.utils.Point;
+
+/** Direction enum for the four cardinal directions. */
+public enum Direction {
+  /** The direction up. */
+  UP("oben", 0, 1),
+  /** The direction down. */
+  DOWN("unten", 0, -1),
+  /** The direction left. */
+  LEFT("links", -1, 0),
+  /** The direction right. */
+  RIGHT("rechts", 1, 0);
+
+  private final String dirName;
+  private final int x;
+  private final int y;
+
+  Direction(String name, int x, int y) {
+    this.dirName = name;
+    this.x = x;
+    this.y = y;
+  }
+
+  /**
+   * Get the name of the direction.
+   *
+   * @return The name of the direction.
+   */
+  public String dirName() {
+    return dirName;
+  }
+
+  /**
+   * Get the x coordinate of the direction.
+   *
+   * @return The x coordinate.
+   */
+  public int x() {
+    return x;
+  }
+
+  /**
+   * Get the y coordinate of the direction.
+   *
+   * @return The y coordinate.
+   */
+  public int y() {
+    return y;
+  }
+
+  /**
+   * Apply the direction to a point.
+   *
+   * <p>When the x coordinate of the direction is not 0, the x coordinate of the point will be moved
+   * by the maximum integer value. When the y coordinate of the direction is not 0, the y coordinate
+   * of the point will be moved by the maximum integer value.
```

Comment:
Der Javadoc liest eigentlich nur den Code auf English vor. Wichtig wäre: Was soll hier passieren und warum?

- Wenn Up/Down -> this.x==0 -> point.x
- Wenn Left -> this.x==-1 -> -infty
- Wenn Right -> this.x==+1 -> +infty

Dito für die y-Koordinate. 

Ich frage mich, warum Du das in einer kryptisch anmutenden Berechnung verbirgst und nicht einfach im Enum bereits -infty bzw. +infty setzt und hier nur die Fallunterscheidung machen musst?

Eine zweite Beobachtung: Sobald ein Teil der Koordinate 0 ist (z.B. x), dann wird für den anderen automatisch der Wert aus dem Argument übernommen.

Ich frage mich, ob hier nicht ein switch/case nach der Konstanten und dann das direkte Erzeugen des Punktes verständlicher ist. Dann kann auch der Ctor weg.

Reply:
Habe die Methode entfernt und einfach inline applied.

> [...] Dann kann auch der Ctor weg.
x & y brauche ich noch und wird auch später für andere Logik verwendet.